src/libexpr/eval.cc

```diff
@@ -1504,9 +1505,17 @@ bool EvalState::isFunctor(Value & fun)
 
 void EvalState::forceFunction(Value & v, const Pos & pos)
 {
+    auto start = std::chrono::high_resolution_clock::now();
+
     forceValue(v);
     if (v.type != tLambda && v.type != tPrimOp && v.type != tPrimOpApp && !isFunctor(v))
         throwTypeError("value is %1% while a function was expected, at %2%", v, pos);
+
```

Comment:
Originally I tried to do an entry time log entry and an exit time log entry to make a stack graph / flame graph, but found the evaluation log messages were linearized, each entry was immediately followed with its exit time -- so obviously this was the wrong place to put that message.